pkg/lua/contrib/lua_loadlib.c

```diff
@@ -0,0 +1,280 @@
+                                                = _luaR_builtin_lua_table;
+
+const size_t luaR_builtin_lua_table_len WEAK = 0;
+
+static const struct luaR_builtin_c _luaR_builtin_c_table[1];
+const struct luaR_builtin_c *const luaR_builtin_c_table WEAK
+                                                = _luaR_builtin_c_table;
+
+const size_t luaR_builtin_c_table_len WEAK = 0;
+
+/* ======================== 'searchers' functions =========================== */
+
+static int _ll_searcher_builtin_lua(lua_State *L, const char *name)
+{
+    const struct luaR_builtin_lua *lmodule =
+        BINSEARCH_STR_P(luaR_builtin_lua_table,
+                        luaR_builtin_lua_table_len,
+                        name, name, LUAR_MAX_MODULE_NAME);
+
+    if (lmodule != NULL) {
+        int load_result = luaL_loadbuffer(L, lmodule->code,
+                                          lmodule->code_size,
+                                          lmodule->name);
+        if (load_result == LUA_OK) {
+            lua_pushstring(L, name);    /* will be 2nd argument to module */
+        }
+
+        return load_result;
+    } else {
+        return LUAR_MODULE_NOTFOUND;
+    }
+}
+
+/**
+ * Search in the list of pure lua modules.
+ *
+ * If the module is found, the source code is compiled and the compiled chunk
+ * is returned.
+ */
+static int searcher_builtin_lua(lua_State *L)
+{
+    const char *name = luaL_checkstring (L, 1);
+    int load_result = _ll_searcher_builtin_lua(L, name);
+
+    switch (load_result) {
+    case LUA_OK:
+        return 2; /* there are two elements in the stack */
+    case LUAR_MODULE_NOTFOUND:
+        return luaL_error(L, "Module '%s' not found in Lua-builtins",
+                          lua_tostring(L, 1));
+    default:
+        return luaL_error(L, "error loading module '%s' from Lua-builtins: \n%s",
+                          lua_tostring(L, 1), lua_tostring(L, 2));
+    }
+}
+
+static int _ll_searcher_builtin_c(lua_State *L, const char *name)
+{
+    const struct luaR_builtin_c *cmodule =
+        BINSEARCH_STR_P(luaR_builtin_c_table,
+                        luaR_builtin_c_table_len,
+                        name, name, LUAR_MAX_MODULE_NAME);
+
+    if (cmodule != NULL) {
+        lua_pushcfunction(L, cmodule->luaopen);
+        lua_pushstring(L, name);    /* will be 2nd argument to module */
+        return LUA_OK;
+    } else {
+        return LUAR_MODULE_NOTFOUND;
+    }
+}
+
+/**
+ * Search in the list of C lua modules.
+ *
+ * If the module is found, the loader function is loaded with lua_pushcfunction
+ * and is returned.
+ */
+static int searcher_builtin_c(lua_State *L)
+{
+    const char *name = luaL_checkstring(L, 1);
+    int load_result = _ll_searcher_builtin_c(L, name);
+
+    if (load_result == LUA_OK) {
+        return 2;
+    } else {
+        return luaL_error(L, "Module '%s' not found in C-builtins",
+                          lua_tostring(L, 1));
+    }
+}
+
+int luaR_getloader(lua_State *L, const char *name)
```

Comment:
Please could you add Doxygen comments to all the functions that have been added to the Lua C API?

Reply:
See the header file.